tests/execute/reboot/data/test.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+REBOOT_COUNT=${REBOOT_COUNT:-0}
```

Comment:
Isn't it a bug if the variable is not defined?

Reply:
@antonvoznia here may your problem lie: try adding a line such as this one to your script.

Reply:
Well, as for now we export the reboot count variable only when reboot happens. But it might make sense to export it always to make it easier for test scripts. On the other hand, for the tests it's always better make good variable checks. @FrNecas, @lukaszachy, thoughts?

Reply:
Well, current scripts running in Beaker do not need this check (sorry, I do not have any public example available), so if you want them to change this behavior, you should not claim that it is compatible.
It might be desirable to have variable checks in scripts, but when an expected variable is not found, the script should immediately abort, not try to hide the problem by setting the variable to some value.

Reply:
IMO test should not create this variable - it is always present in [Beaker](https://beaker-project.org/docs/user-guide/task-environment.html#environment-variables)
We create reboot scripts each time so we should set the count as well.

So I'd say missing variables should be an instant test failure (of reboot functional test)

Reply:
Yes, I think missing variable should be taken to indicate missing reboot support (not that we have not rebooted yet). It might be a good idea to leave it unset for provisioners / executors that do not support reboot (like the local provisioner).